File: app/src/main/java/fi/kaupunkifillarit/analytics/ErrorEvents.java
package fi.kaupunkifillarit.analytics;

import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.common.GoogleApiAvailability;

import java.util.Map;

public abstract class ErrorEvents {
    private static final String CATEGORY = "Error";

    public static Map<String, String> playServicesError(String error) {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction("Play Services")
                .setLabel(error)
                .set("Version", String.valueOf(GoogleApiAvailability.GOOGLE_PLAY_SERVICES_VERSION_CODE))
                .setValue(1)
                .build();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/analytics/ErrorEvents.kt
package fi.kaupunkifillarit.analytics

import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.common.GoogleApiAvailability

object ErrorEvents {
    private val CATEGORY = "Error"

    fun playServicesError(error: String): Map<String, String> {
        return HitBuilders.EventBuilder().setCategory(CATEGORY).setAction("Play Services").setLabel(error).set("Version", GoogleApiAvailability.GOOGLE_PLAY_SERVICES_VERSION_CODE.toString()).setValue(1).build()
    }
}
